### ISSUE TITLE
Re-use keep-alive HTTP connections across clients

### ISSUE THREAD

**Issue by anujm08** [Azure/go-autorest#554]
For every new autorest client created, autorest creates a new HTTP Transport and Client at https://github.com/Azure/go-autorest/blob/master/autorest/sender.go#L109. This does not allow keep-alive HTTP connections to be re-used across different clients and since the keep-alive HTTP connections keep the 2 goroutines running (for any possible re-use) for about 90 seconds, this leads to execessive memory usage if a lot of clients are being created. 

Can we provide an option to re-use the same autorest Transport variable across different initializations of Client which will allow possible re-use of the alive HTTP connections? 

I also noticed that this behaviour got changed in https://github.com/Azure/go-autorest/commit/5e7a399d8bbf4953ab0c8e3167d7fd535fd74ce1 and before that we were using the Transport variable defined in tracing package across all client initializations.

We can see the effect of this issue from the following sample code. The number of goroutines count keep increasing by 2 after each iteration.
```
// authorizer := <bearer authorizer>

for i := 0; i < 10; i++ {
	client := compute.NewVirtualMachinesClient(subscriptionID)
	client.Authorizer = authorizer
	client.ListAllComplete(ctx)
	log.Printf("Number of goroutines: %d", runtime.NumGoroutine())
}
```

**Comment by jhendrixMSFT** [comment]
I have a fix for this, it will cache senders per TLS renegotiation type.

**Comment by jhendrixMSFT** [comment]
Fixed in module `autorest/v0.11.9`
